### PR TITLE
try fix freebsd

### PR DESCRIPTION
**Pull request by IanButterworth**
Trying to fix downstream issue:
```
dlopen("/root/.julia/packages/VideoIO/zSsdl/deps/usr/lib/libavformat.so")
ERROR: could not load library "/root/.julia/packages/VideoIO/zSsdl/deps/usr/lib/libavformat.so"
Shared object "libstdc++.so.6" not found, required by "libvpx.so.6"
```

### CODE REVIEW

File: build_tarballs.jl [hunk 1]
```diff
@@ -49,7 +49,7 @@ elif [[ "${target}" == x86_64-w64-mingw32 ]]; then
 elif [[ "${target}" == x86_64-unknown-* ]]; then
```
Comment (staticfloat) [inline build_tarballs.jl]: I would compare against `*freebsd*`, not `x86_64-unknown-*`, as that could match many non-FreeBSD things.

File: build_tarballs.jl [hunk 2]
```diff
@@ -49,7 +49,7 @@ elif [[ "${target}" == x86_64-w64-mingw32 ]]; then
...
     export CROSS=x86_64-unknown-gcc
     export CC=/opt/${target}/bin/${target}-gcc
-    export CXX=/opt/${target}/bin/${target}-g++
+    export CXX=/opt/${target}/bin/${target}-clang++
```
Comment (staticfloat) [inline build_tarballs.jl]: `CC` and `CXX` should already be set appropriately; have you tried just not setting them?

Comment (staticfloat) [reply build_tarballs.jl]: If you have, what breaks?  In order to use `clang` you need to provide `-target` and `--sysroot` arguments; here's what the defaults are for targeting FreeBSD:

```
# echo $CC
ccache /opt/x86_64-linux-gnu/tools/clang -target x86_64-unknown-freebsd11.1 --sysroot /opt/x86_64-unknown-freebsd11.1/x86_64-unknown-freebsd11.1/sys-root
```

**Review by staticfloat — CHANGES_REQUESTED** [review]

**Comment by IanButterworth** [comment]
This build was mostly @giordano's effort, so I'm sure he can explain the logic above better than me

**Comment by staticfloat** [comment]
Okay so I tried removing all the `CC`, `CXX` and `LD` entries for FreeBSD, and it built just fine for me.  I'll wait for @giordano to weigh in, but my guess is that that should just work.

**Comment by giordano** [comment]
I used GCC instead of Clang because in the [README](https://github.com/webmproject/libvpx/blob/13dc6d671ccd88a8af021b1b682400bf16b6c7c6/README) they say they support GNU toolchan only, with other POSIX toolchains *likely* to work.  For macOS, the only target that worked was "x86_64-darwin14-gcc", without the "-gcc" I couldn't make it work (probably it isn't recognised) and thus used GCC as a consequence.  For FreeBSD I used GCC just following the example of what I did for macOS.

I just verified that simply removing all `CC`, `CXX`, `LD` settings, for both macOS and FreeBSD, makes compilation succeed on both platform, however I can't test myself if the libraries produced do work.  That could be worth trying

File: build_tarballs.jl (hunk 1 above)
Comment (giordano) [reply build_tarballs.jl]: Yep, I agree, I realised that after the PR.

**Comment by staticfloat** [comment]
Perfect, so let's just lie, tell the buildsystem we're using GNU tools, then do whatever the heck we want.

**Comment by IanButterworth** [comment]
I just noticed that freebsd is missing a `export TARGET=$CROSS` in master.. could that contribute?

I'm trying a build with removing `CC`, `CXX`, `LD` and adding `export TARGET=$CROSS`

**Comment by IanButterworth** [comment]
It's fixed! VideoIO is working on freebsd! https://github.com/JuliaIO/VideoIO.jl/pull/180

For the record, I built freebsd and apple, and updated those bins at https://github.com/JuliaIO/LibVPXBuilder/releases/tag/v1.8.0

**Comment by giordano** [comment]
> I just noticed that freebsd is missing a `export TARGET=$CROSS` in master.. could that contribute?

No, because there is a default
https://github.com/JuliaIO/LibVPXBuilder/blob/a3730a245d550f1481ebb13bb22020ad3d978fa0/build_tarballs.jl#L26
for all platforms :wink:

**Pull request merged**